src/lib.rs

```diff
@@ -1098,6 +1098,21 @@ mod test {
     }
 
     #[test]
+    #[should_panic(expected = "Unexpected value")]
```

Comment:
The panic is occurring because this test is calling `unwrap` on an Err value, which is what is returned by the deserializer when an integer is not present.